byterun/major_gc.c

```diff
@@ -554,3 +555,18 @@ void caml_init_major_heap (asize_t heap_size)
   caml_allocated_words = 0;
   caml_extra_heap_resources = 0.0;
 }
+
+void caml_finalise_heap (void)
+{
+  // Finishing major cycle (all values become white)
+  caml_empty_minor_heap ();
+  caml_finish_major_cycle ();
+  Assert (caml_gc_phase == Phase_idle);
+
+  // Finalising all values (by forced sweeping)
```

Comment:
Understood: this is needed to prepare the free-list for the sweep phase. Fixed it.